modules/UI/videolayout/RemoteVideo.js

```diff
@@ -189,35 +160,28 @@ RemoteVideo.prototype._generatePopupContent = function () {
     const participantID = this.id;
 
     /* jshint ignore:start */
-    return (
-        <RemoteVideoMenu id = { participantID }>
-            { isModerator
-                ? <MuteButton
+    ReactDOM.render(
+        <Provider store = { APP.store }>
+            <I18nextProvider i18n = { i18next }>
+                <RemoteVideoMenuTriggerButton
+                    initialVolumeValue = { initialVolumeValue }
                     isAudioMuted = { this.isAudioMuted }
-                    onClick = { this._muteHandler }
-                    participantID = { participantID } />
-                : null }
-            { isModerator
-                ? <KickButton
-                    onClick = { this._kickHandler }
-                    participantID = { participantID } />
-                : null }
-            { remoteControlState
-                ? <RemoteControlButton
-                    onClick = { onRemoteControlToggle }
+                    isModerator = { isModerator }
+                    onMenuDisplay = { this._onRemoteVideoMenuDisplay.bind(this) }
+                    onRemoteControlToggle = { onRemoteControlToggle }
+                    onVolumeChange = { onVolumeChange }
                     participantID = { participantID }
                     remoteControlState = { remoteControlState } />
-                : null }
-            { onVolumeChange
-                ? <VolumeSlider
-                    initialValue = { initialVolumeValue }
-                    onChange = { onVolumeChange } />
-                : null }
-        </RemoteVideoMenu>
-    );
+            </I18nextProvider>
+        </Provider>,
+        remoteVideoMenuContainer);
```

Comment:
The piece of dom that used to be hovered over to display the remote menu was not in react yet. It was necessary to put that into react for InlineDialog. That piece of dom was moved into RemoteVideoMenuTriggerButton.